lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -519,72 +613,110 @@ def _rows_for_users(self, context, users):
+            return ProblemGradeReport()._generate(context)
+
+    def _generate(self, context):
         """
         Generate a CSV containing all students' problem grades within a given
         `course_id`.
         """
+        context.update_status(u'Starting grades')
+        course = get_course_by_id(context.course_id)
+        header_row = OrderedDict([('id', 'Student ID'), ('email', 'Email'), ('username', 'Username')])
+        graded_scorable_blocks = self._graded_scorable_blocks_to_header(course)
+        success_headers = self._success_headers(header_row, graded_scorable_blocks)
+        error_headers = self._error_headers(header_row)
 
-        def log_task_info(message):
-            """
-            Updates the status on the celery task to the given message.
-            Also logs the update.
-            """
-            fmt = u'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
-            task_info_string = fmt.format(
-                task_id=task_id, entry_id=_entry_id, course_id=course_id, task_input=_task_input
-            )
-            TASK_LOG.info(u'%s, Task type: %s, %s, %s', task_info_string, action_name, message, task_progress.state)
+        generated_rows = self._batched_rows(context, header_row, graded_scorable_blocks)
+        success_rows, error_rows = zip(*generated_rows)
+        success_rows = list(chain(*success_rows))
+        error_rows = list(chain(*error_rows))
 
-        start_time = time()
-        start_date = datetime.now(UTC)
-        status_interval = 100
-        task_id = _xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None
+        # update metrics on task status
+        context.task_progress.succeeded = len(success_rows)
+        context.task_progress.failed = len(error_rows)
+        context.task_progress.attempted = context.task_progress.succeeded + context.task_progress.failed
+        context.task_progress.total = context.task_progress.attempted
 
-        report_for_verified_only = generate_grade_report_for_verified_only()
-        enrolled_students = CourseEnrollment.objects.users_enrolled_in(
-            course_id=course_id,
-            include_inactive=True,
-            verified_only=report_for_verified_only,
-        )
-        task_progress = TaskProgress(action_name, enrolled_students.count(), start_time)
+        context.update_status(u'Completed grades')
 
-        # This struct encapsulates both the display names of each static item in the
-        # header row as values as well as the django User field names of those items
-        # as the keys.  It is structured in this way to keep the values related.
-        header_row = OrderedDict([('id', 'Student ID'), ('email', 'Email'), ('username', 'Username')])
+        date = datetime.now(UTC)
+        upload_csv_to_report_store([success_headers] + success_rows, 'grade_report', context.course_id, date)
```

Comment:
Please correct the report name, its `problem_grade_report` and `problem_grade_report_err`